### PR TITLE
implemented pecs while resolving task

### PR DESCRIPTION
**Pull request by happytroya**

### CODE REVIEW

File: src/main/java/core/mate/academy/model/BulldozerProducer.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package core.mate.academy.model;
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/model/BulldozerProducer.java:1]:
```suggestion
package core.mate.academy.service;
```

File: src/main/java/core/mate/academy/model/ExcavatorProducer.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package core.mate.academy.model;
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/model/ExcavatorProducer.java:1]:
```suggestion
package core.mate.academy.service;
```

File: src/main/java/core/mate/academy/model/TruckProducer.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package core.mate.academy.model;
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/model/TruckProducer.java:1]:
```suggestion
package core.mate.academy.service;
```

File: src/main/java/core/mate/academy/model/BulldozerProducer.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+import core.mate.academy.service.MachineProducer;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Bulldozer> {
+    @Override
+    public List<Bulldozer> get() {
+        return List.of(new Bulldozer[]{new Bulldozer()});
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/model/BulldozerProducer.java:9]:
```suggestion
        return List.of(new Bulldozer());
```

File: src/main/java/core/mate/academy/model/ExcavatorProducer.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+import core.mate.academy.service.MachineProducer;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
+    @Override
+    public List<Excavator> get() {
+        return List.of(new Excavator[]{new Excavator()});
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/model/ExcavatorProducer.java:9]:
```suggestion
        return List.of(new Excavator());
```

File: src/main/java/core/mate/academy/model/TruckProducer.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+import core.mate.academy.service.MachineProducer;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
+    @Override
+    public List<Truck> get() {
+        return List.of(new Truck[]{new Truck()});
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/model/TruckProducer.java:9]:
```suggestion
        return List.of(new Truck());
```

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
 /**
  * Parameterize this service and add its implementation in a separate class.
  */
-public interface MachineService {
-
+public interface MachineService<T> {
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/MachineService.java:9]:
```suggestion
public interface MachineService<T extends Machine> {
```

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 8]
```diff
@@ -26,13 +26,14 @@ public interface MachineService {
...
+
 
     /**
      * Call the method doWork() from each machine.
      * This method should be able to accept a list of bulldozers as well as list of trucks.
      *
      * @param machines - the list of machines
      */
-    void startWorking(List<Object> machines);
+    void startWorking(List<? extends Machine> machines);
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/MachineService.java:38]:
```suggestion
    void startWorking(List<? extends T> machines);
```

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 9]
```diff
@@ -1,7 +1,48 @@
...
+    private BulldozerProducer bulldozerProducer = new BulldozerProducer();
+    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private TruckProducer truckProducer = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+        if (type.equals(Bulldozer.class)) {
+            machines = bulldozerProducer.get();
+            return new ArrayList<>(machines);
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:26]:
```suggestion
            return new ArrayList<>(bulldozerProducer.get());
```

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 10]
```diff
@@ -1,7 +1,48 @@
...
+public class MachineServiceImpl implements MachineService<Machine> {
+    private BulldozerProducer bulldozerProducer = new BulldozerProducer();
+    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private TruckProducer truckProducer = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+        if (type.equals(Bulldozer.class)) {
+            machines = bulldozerProducer.get();
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:25]:
```suggestion
        if (type.equals(Bulldozer.class)) {
```

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 11]
```diff
@@ -1,7 +1,48 @@
...
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        for (int i = 0; i < machines.size(); i++) {
+            machines.set(i, value);
+        }
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
+
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:46]: complete this method, pls :)

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]
Good job

**Pull request closed without merging**